src/SprykerEco/Zed/Econda/Business/Exporter/Writer/File/Adapter/CsvAdapter.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Econda\Business\Exporter\Writer\File\Adapter;
+
+use SplFileObject;
+use SprykerEco\Zed\Econda\Business\Exporter\Exception\FileWriterException;
+use SprykerEco\Zed\Econda\EcondaConfig;
+
+class CsvAdapter implements AdapterInterface
+{
+    /**
+     * @var string
+     */
+    protected $folderPath;
+
+    /**
+     * @var string
+     */
+    protected $fileName;
+
+    /**
+     * @var string
+     */
+    protected $delimiter;
+
+    /**
+     * @var string
+     */
+    protected $enclosure;
+
+    /**
+     * @var string
+     */
+    protected $escape;
+
+    /**
+     * @var \SplFileObject|null
+     */
+    protected $csvFile;
+
+    /**
+     * @param string $directory
+     * @param string $delimiter
+     * @param string $enclosure
+     * @param string $escape
+     */
+    public function __construct($directory, $delimiter = ",", $enclosure = "\"", $escape = "\\")
+    {
+        $this->folderPath = $directory;
+
+        $this->delimiter = $delimiter;
+        $this->enclosure = $enclosure;
+        $this->escape = $escape;
+    }
+
+    /**
+     * @param array $data
+     * @param string $type
+     *
+     * @return bool
+     */
+    public function write(array $data, $type = '')
+    {
+        $result = false;
+        $csvFile = $this->getCsvFile($data);
+        foreach ($data as $key => $row) {
+            $result = $csvFile->fputcsv($row, EcondaConfig::ECONDA_CSV_DELIMITER) !== false ? true : false;
```

Comment:
In this case, if **all** `rows` will fail, only the **last** one will be successful — `true` will be returned… is this an expected behavior?